cookbook-react-native/recipes/images/loadable-image/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { ActivityIndicator, Image, View, ViewStyle, ImageResizeMode } from 'react-native';
+import FastImage, { Source, FastImageProps } from 'react-native-fast-image';
+
+import styles from './styles';
+
+interface Props extends FastImageProps {
+  activityIndicatorColor?: string;
+  backgroundColor?: string;
+  placeholder?: number;
+  placeholderResizeMode?: ImageResizeMode;
+  source: Source;
+  style?: ViewStyle;
+}
+
+function LoadableImage({
+  activityIndicatorColor,
+  backgroundColor,
+  placeholder,
+  placeholderResizeMode,
+  source,
+  onError,
+  onLoadEnd,
+  style,
+  ...props
+}: Props) {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const handleLoaded = () => {
+    onLoadEnd?.();
+    setLoading(false)};
+
+  const handleError = () => {
+    onError?.();
+    setError(true);
+  };
+
+  // If you want to have a default image placeholder, remove the conditional to show the image
+  // and replace the source with `placeholder || imagePlaceholder`
+  return (
+    <View style={[style, styles.container]}>
+      {(loading || error || !source) && (
+        <View style={[styles.contentContainer, styles.fill, { backgroundColor }]}>
+          {!!placeholder && <Image source={placeholder} style={styles.fill} resizeMode={placeholderResizeMode} />}
+          {loading && source?.uri && (
```

Comment:
Why do you need the `source?.uri` here?

Reply:
I ask myself the same thing when I copied this component and I believe that if we don't have a `source.uri` (local asset) it would be load quite fast and we will barely get to see the activity indicator. I guess.

Anyway, I think I would keep the condition